src/mlpack/bindings/python/mlpack/pythonize_model.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+#
+# Utilities for model parameters extraction.
+#
+# @file pythonize_model.py
+# @author Mehul Kumar Nirala
+#
+# mlpack is free software; you may redistribute it and/or modify it under the
+# terms of the 3-clause BSD license.  You should have received a copy of the
+# 3-clause BSD license along with mlpack.  If not, see
+# http://www.opensource.org/licenses/BSD-3-Clause for more information.
+
+
+import numpy as np
+from collections import defaultdict
+from xml.etree import cElementTree as ET
+
+def etree_to_dict(t, attrib = False):
+    d = {t.tag: {} if t.attrib else None}
+    children = list(t)
+    if children:
+        dd = defaultdict(list)
+        for dc in map(etree_to_dict, children):
+            for k, v in dc.items():
+                dd[k].append(v)
+        d = {t.tag: {k:v[0] if len(v) == 1 else v for k, v in dd.items()}}
+    if t.attrib and attrib:
+        d[t.tag].update(('@' + k, v) for k, v in t.attrib.items())
+    if t.text:
+        text = t.text.strip()
+        if children or t.attrib:
+            if text:
+              d[t.tag]['#text'] = text
+        else:
+            d[t.tag] = text
+    return d
+
+def xml_to_dict(s):
+    tree = ET.XML(s)
+    d = etree_to_dict(tree)
+    d = d[d.keys()[0]]
+    d = d[d.keys()[0]]
+    return d
+
+def transform(model):
+    result = {}
+    try:
+        if type(model).__name__ != 'str':
+            model = model.__getparams__()
+    except Exception as e:
+        raise e
+    
+    model_dict = xml_to_dict(model)
+
+    # network parameters
+    try:
+        result["parameters"] = model_dict["parameters"]["item"]
```

Comment:
Right, so to respond to this:

> @rcurtin That's exactly what I wanted to say. We cannot get a lot of information from the serialize(). However I tried to extract as many information as possible in the PR.

I agree completely, but what I think that we should do here in `transform()` is try and take a look at what we have in the dict, and label it accordingly.  For instance, if we detect the Armadillo serialization members (like `n_rows`, `n_cols`, `n_elem`, etc.) we should replace that part of the dict with a numpy `ndarray`.  You can take a look at `src/mlpack/core/arma_extend/Mat_extra_meat.hpp` to see the matrix serialization details.  (There is also serialization for sparse matrices in `src/mlpack/core/arma_extend/SpMat_extra_meat.hpp` and we can use that to convert the corresponding dict element to a scipy sparse matrix or something.)

Reply:
Apologies for the delay. I have incorporated the changes